rsa.js

```diff
@@ -20,26 +20,72 @@ function sanitize(key) {
   return key.replace(new RegExp('\n|\r', 'g'), '');
 }
 
+function removeEndingEquals(blob) {
+  if (blob.endsWith('==')) {
+    return blob.slice(0,-2);
+  } else if (blob.endsWith('=')) {
+    return blob.slice(0,-1);
+  }
+  return blob;
+}
+
+function illegalBase64(blob) {
```

Comment:
I think that checking that the blob matches `[A-Za-z0-9/+]+={0,2}` would considerably simplify this piece of code while being good enough (you lose the distinction between "bad padding" and "wrong character" but if the key is invalid there's nothing much you can do anyway).